serverlessworkflow/modules/ROOT/pages/eventing/working-with-openapi-callbacks.adoc

```diff
@@ -0,0 +1,117 @@
+= Openapi Callback in {context}
+
+:compat-mode!:
+// Metadata:
+:description: Working with openapi callbacks
+:keywords: kogito, workflow, serverless, openapi, callback, event
+
+This document describes the link:{open_api_spec_url}#callback-object[Openapi Callback] and related examples. Also it covers the difference between Openapi Callback and the Callback state. Both openapi callback and the Callback state performs an action and waits for an event, to be produced as a result of the action, to resume the workflow. Therefore, both openapi callback and the Callback state are suitable to perform  _fire&wait-for-result_ operations. The openapi callback is an asynchronous, out-of-band requests that your service will send to some other service in response to certain events whereas the action peformed by a Callback state is an asynchronous external service invocation. The goal of both the ways is to send an event to workflow to resume the flow.
+
+As mentioned in the Openapi Callback document, while the workflow performs an asynchronous service invocation callback URL must be provided in the request body. The callback which sends the event to the workflow is defined using the URL provided by the user during service invocation.
+
+From a workflow perspective, asynchronous service indicates that the control is returned to the caller immediately without waiting for the action to be completed. Once the action is completed, a link:{cloud_events_url}[CloudEvent] is published to resume the workflow.
+
+For the workflow to identify the published CloudEvent it is waiting for, the external service developer includes the workflow instance ID in the CloudEvent header or uses the xref:eventing/event-correlation-with-workflows.adoc[Event correlation]. The following figure displays the process:
+
+image::eventing/callbacks-explained.png[]
+
+The workflow correlation described in this document focuses on the former mechanism that is based on the fact that each workflow instance contains a unique identifier, which is generated automatically.
+
+[[ref-example-callback]]
+== Example of Openapi Callback
+
+To understand the Openapi Callback, see the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus[`serverless-workflow-callback-events-over-http-quarkus`] example application in GitHub repository.
+
+The `serverless-workflow-callback-events-over-http-quarkus` application is initiated with the below request:
+[code,json]
+----
+{
+  "message": "Hello"
+}
+----
+Once the workflow is started, it makes an external service call with the callback URL and process instance Id in the request body. As configured in the openapi file, the callback URL is invoked to send a CloudEvent to the workflow.
+
+Once the `wait` type CloudEvent is received, the workflow moves to the next state and ends successfully.
```

Comment:
I think it's worth bringing the workflow visual representation to the guide or the workflow DSL.